### PR TITLE
input_common: Allow disabling GCAdapter on platforms without libusb.

### PR DESCRIPTION
**Pull request by Steveice10**
Some platforms like iOS are not supported by libusb. Adding an upstream build option to disable libusb (and by extension GameCube Adapter support) to ease porting to these platforms.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6448)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +31,8 @@ option(ENABLE_WEB_SERVICE "Enable web services (telemetry, etc.)" ON)
 
 option(ENABLE_CUBEB "Enables the cubeb audio backend" ON)
 
+CMAKE_DEPENDENT_OPTION(ENABLE_LIBUSB "Enable libusb for GameCube Adapter support" OFF "IOS" ON)
```
Comment (FearlessTobi) [inline CMakeLists.txt:34]: do we support ios?

**Review by FearlessTobi — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (Steveice10) [reply CMakeLists.txt:34]: Not currently but it’s an example of a platform that needs it, if nothing else to set up the skeleton for platforms to turn it off here.

And I was under the impression that passing in this order was on by default, but from the docs it looks like the ON case here would be a local variable and not an option… Let me fix that up real quick

File: src/input_common/CMakeLists.txt [hunk 2]
```diff
@@ -33,10 +29,20 @@ if(ENABLE_SDL2)
...
 
+if(ENABLE_LIBUSB)
+    target_sources(input_common PRIVATE
+        gcadapter/gc_adapter.cpp
+        gcadapter/gc_adapter.h
+        gcadapter/gc_poller.cpp
+        gcadapter/gc_poller.h
+    )
+    target_include_directories(input_common PRIVATE ${LIBUSB_INCLUDE_DIR})
+    target_link_libraries(input_common PUBLIC ${LIBUSB_LIBRARIES})
```
Comment (SachinVin) [inline src/input_common/CMakeLists.txt:40]: Could this also be `PRIVATE` ?

Comment (Steveice10) [reply src/input_common/CMakeLists.txt:40]: Potentially but this diff is just indenting what already exists to wrap it in the enable flag.

Comment (SachinVin) [reply src/input_common/CMakeLists.txt:40]: Might as well fix it since we are touching it

**Comment by Steveice10** [comment]
Rebased and addressed comment

**Comment by SachinVin** [comment]
Thanks

**Pull request merged**